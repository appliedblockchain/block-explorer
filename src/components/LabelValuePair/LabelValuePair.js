import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import styles from './style.module.css'

const LabelValuePair = ({ label, isCode, children }) =>
  <div className={styles.labelValuePair}>
    <div className={styles.labelValuePairItem}>
      <div className={styles.labelValuePairLabel}>{label}</div>
      <div className={classNames(styles.labelValuePairValue, isCode && styles.codeValue)}>
        {children}
      </div>
    </div>
  </div>

LabelValuePair.propTypes = {
  label: PropTypes.string,
  isCode: PropTypes.bool,
  children: PropTypes.node
}

export default LabelValuePair
