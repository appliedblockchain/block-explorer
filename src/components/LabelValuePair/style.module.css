
.labelValuePair {
    padding: 10px;
    box-sizing: border-box;
    width: 100%;
}

.labelValuePairItem {
    display: flex;
}

.labelValuePairLabel {
    font-weight: bold;
    width: 140px;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
}

.labelValuePairValue {
    flex: 5;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
}

.codeValue {
    flex: 5;
    word-break: break-all;
    white-space: normal;
    font-size: 14px;
    font-family: 'Inconsolata';
}