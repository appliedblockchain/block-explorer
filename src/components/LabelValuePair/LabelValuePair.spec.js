import React from 'react'
import renderer from 'react-test-renderer'
import LabelValuePair from './LabelValuePair'

it('renders correctly', () => {
  const tree = renderer
    .create(
      <LabelValuePair label="Label">
        <div>Value children</div>
      </LabelValuePair>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})
