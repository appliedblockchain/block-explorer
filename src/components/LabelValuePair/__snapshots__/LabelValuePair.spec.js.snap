// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="labelValuePair"
>
  <div
    className="labelValuePairItem"
  >
    <div
      className="labelValuePairLabel"
    >
      Label
    </div>
    <div
      className="labelValuePairValue"
    >
      <div>
        Value children
      </div>
    </div>
  </div>
</div>
`;
